x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpClient.java

```diff
@@ -92,7 +92,7 @@ public void completed(HttpResponse response) {
 
             @Override
             public void failed(Exception ex) {
-                throttlerManager.getThrottler().warn(logger, format("Request [%s] failed", request.getRequestLine()), ex);
+                throttlerManager.warn(logger, format("Request [%s] failed", request.getRequestLine()), ex);
```

Comment:
Just noticed this PR and I'm wondering: why do we implement our own HTTP client and our own retry logic? Aren't there solutions for that, that work out of the box?

Reply:
I also fail tot understand what this `HttpClient` does on top of the Apache `CloseableHttpAsyncClient` (a little bit of class-level JavaDoc wouldn't hurt.

Reply:
> Just noticed this PR and I'm wondering: why do we implement our own HTTP client and our own retry logic? Aren't there solutions for that, that work out of the box?

Yeah great feedback, I'm in the process of refactoring the retry class to leverage the `RetryableAction` that Dave suggested.

The apache http client does support adding retry logic but it's on a later library version 😞 . We could certainly upgrade but there are other parts of elasticsearch that rely on the http client so I was worried about breaking changes. I'm sure there's some gradle magic we could use to only apply the http client version to this plugin time hasn't really allowed for digging into that.

> I also fail to understand what this HttpClient does on top of the Apache CloseableHttpAsyncClient (a little bit of class-level JavaDoc wouldn't hurt

Yeah good point, I'll add some javadoc. This client handles setting up the client and returning the response on a separate thread from the internal apache http client's.